IbFlexReader/IbFlexReader/Reader.cs

```diff
@@ -2,92 +2,215 @@
+				if (TryGetFlexStatementResponse(stream, out FlexStatementResponse flexStatementResponse))
+				{
+					return new FlexResult
+					{
+						Status = flexStatementResponse.Status,
+						ErrorCode = flexStatementResponse.ErrorCode,
+						ErrorMessage = flexStatementResponse.ErrorMessage
+					};
+				}
+
+                var queryResponse = Deserializer.Deserialize<Xml.Contracts.FlexQueryResponse, Contracts.FlexQueryResponse>(stream, out var errorObjects);
+
+                // in case of an error during deserialization, queryResponse maybe NULL (see PopulateFrom method)
+                if (queryResponse == null)
                 {
-                    if (dumpFile != null) { Utils.DumpFileWriter.DumpGetStatement(stream, dumpFile); }
-
-                    var queryResponse = Deserializer.Deserialize<Xml.Contracts.FlexQueryResponse, Contracts.FlexQueryResponse>(stream, out var errorObjects);
-                    queryResponse.Errors = errorObjects;
-
-                    return new FlexResult
-                    {
-                        Status = "Success",
-                        FlexQueryResponse = queryResponse
-                    };
-                }
-            }
-        }
-    }
+					if (errorObjects.Count > 0)
+					{
+						System.Text.StringBuilder sb = new System.Text.StringBuilder();
+						errorObjects.ForEach(e => sb.AppendLine(e.Message));
+						throw new Exception(sb.ToString());
+					}
+					else
+					{
+						throw new Exception("deserialization failed");
+					}
+				}
+
+				queryResponse.Errors = errorObjects;
+
+				return new FlexResult
+				{
+					Status = "Success",
+					FlexQueryResponse = queryResponse
+				};
+			}
+		}
+
+		private async Task<FlexResult> SendRequest(HttpClient client, string token, string queryId, string dumpFile)
+		{
+			var uri = new Uri($"https://gdcdyn.interactivebrokers.com/Universal/servlet/FlexStatementService.SendRequest?t={token}&q={queryId}&v=3");
+
+			using (var sendRequestResult = await client.PostAsync(uri, null))
+			using (var stream = await sendRequestResult.Content.ReadAsStreamAsync())
+			{
+				if (dumpFile != null) { Utils.DumpFileWriter.DumpSendRequest(stream, dumpFile); }
+
+				var response = Deserializer.Deserialize<XmlFlexStatementResponse, FlexStatementResponse>(stream, out var errorObjects);
+
+				if (response == null)
+				{
+					if (errorObjects.Count > 0)
+					{
+						System.Text.StringBuilder sb = new System.Text.StringBuilder();
+						errorObjects.ForEach(e => sb.AppendLine(e.Message));
+						throw new Exception(sb.ToString());
+					}
+					else
+					{
+						throw new Exception("deserialization failed");
+					}
+				}
+
+				if (response.Status == "Success")
+				{
+					//referenceCode = response.ReferenceCode.Value.ToString();
+
+					return new FlexResult
+					{
+						Status = "Success",
+						ErrorMessage = response.ReferenceCode.Value.ToString()
```

Comment:
I think we can use a new property on FlexResult, named `ReferenceCode`.

Reply:
Done